### PR TITLE
Fix custom REST query URL validation

### PR DESCRIPTION
**Pull request by melohagan**
## Description
Fixes custom REST query URL validation so the `Send` button uses the same URL validation as `Save`. When a user enters a valid domain without a protocol, the request cannot be sent and a persistent tooltip prompts them to include `http://` or `https://`.

## Addresses
- https://github.com/Budibase/budibase/issues/18798

## Screenshots
PROTOCOL MISSING:
no protocol

PROTOCOL INCLUDED:
with protocol

## Launchcontrol
Custom REST query requests now consistently require URLs to include a protocol before they can be saved or sent.

<a href="https://cubic.dev/pr/Budibase/budibase/pull/19020?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

File: packages/builder/src/components/integration/query.ts [hunk 1]
```diff
@@ -487,3 +487,18 @@ export function isValidEndpointUrl(url: string | undefined): boolean {
...
+export function isValidEndpointUrlMissingProtocol(
+  url: string | undefined
+): boolean {
+  if (!url) return false
+  if (/^(https?:\/\/|\{\{)/.test(url)) return false
+  if (url.startsWith("/") || /\s/.test(url)) return false
+  if (/^[a-z][a-z\d+.-]*:/i.test(url)) return false
+  try {
+    const parsed = new URL(`https://${url}`)
+    return /^[^/?#]+\.[^/?#]+/.test(parsed.hostname)
```
[the remark this answers is not in the document]
Comment (melohagan) [reply packages/builder/src/components/integration/query.ts]: I want the missing-protocol detection to be stricter so that the tooltip only displays to the user specifically when they have a valid URL, but it's just missing the protocol.

File: packages/builder/src/components/integration/CustomEndpointInput.svelte [hunk 2]
```diff
@@ -9,6 +15,7 @@
   export let url: string = ""
   export let baseUrlOptions: { label: string; url: string }[] = []
   export let disabled: boolean = false
+  export let showProtocolWarning: boolean = false
```
Comment (adrinr) [inline packages/builder/src/components/integration/CustomEndpointInput.svelte]: This would be more extensible if instead of a boolean, we passed a "warningText" field. This will allow adding different validation messages in the future

**Review by adrinr — DISMISSED** [review]

File: packages/builder/src/components/integration/CustomEndpointInput.svelte (hunk 2 above)
Comment (melohagan) [reply packages/builder/src/components/integration/CustomEndpointInput.svelte]: Refactored

File: packages/builder/src/components/integration/query.ts [hunk 3]
```diff
@@ -487,3 +487,21 @@ export function isValidEndpointUrl(url: string | undefined): boolean {
...
   }
 }
+
+export function isValidEndpointUrlMissingProtocol(
+  url: string | undefined
+): boolean {
+  if (!url) return false
+  if (/^(https?:\/\/|\{\{)/.test(url)) return false
+  if (url.startsWith("/") || /\s/.test(url)) return false
+  if (/^[a-z][a-z\d+.-]*:\/\//i.test(url)) return false
```
[the remark this answers is not in the document]
Comment (melohagan) [reply packages/builder/src/components/integration/query.ts:497]: I don't think this edge case is worth addressing; not expecting users to enter mailto into the URL bar. 

Additionally, technically you could have a URL like: `https://mailto:password@example.com` where *mailto* is their username.

**Review by deanhannigan — APPROVED** [review]
LGTM! Ran through this locally and all issues were resolved. Nice one! 🚀
